### ISSUE TITLE
lost the 'lib' directory from original version of tree-lstm, lead to parsing errors

### ISSUE THREAD

**Issue by shentao51** [dasguptar/treelstm.pytorch#3]
just copy the 'lib' dir from https://github.com/stanfordnlp/treelstm to this code main dir.

**Comment by dasguptar** [comment]
Hi @shentao51 , if you run `fetch_and_preprocess.py` or `scripts/download.py`, it will create a `lib` directory with the necessary files. Alternatively, you can copy over the files from the [original treelstm repository](https://github.com/stanfordnlp/treelstm).

**Comment by dasguptar** [comment]
@shentao51 
Ah, yes, good catch. Will upload those *.java files from the original repository. Thanks.

**Comment by dasguptar** [comment]
@shentao51 I added the `lib/*.java` files from the original repository. Things should work as expected now. Thanks for pointing out the bug! It has slipped my eyes since I already had those files in my working directory.

**Comment by shentao51** [comment]
Hi @dasguptar, I also found that your version of Tree-LSTM does not support mini batch training, but you added the argparser with arg 'batch size'. I think this maybe make some mislead. I suggest you delete the arg. Thanks a lot for your timely replying.

**Comment by dasguptar** [comment]
Hi @shentao51 , if you look at [these lines](https://github.com/dasguptar/treelstm.pytorch/blob/master/trainer.py#L32-L35), you will see that gradients are accumulated over a mini-batch and weight updates via the optimizer only happen after a mini-batch of sampls have been "seen" by the model. 
This is in effect similar to what the original code is doing, since there is no simple way on how to actually send a mini-batch of samples through a recursive model. (Because the recursive structure will vary for each sample, naive mini-batching breaks)

**Comment by shentao51** [comment]
Oh, I got it, due to the different tree architecture in the minibatch. I appreciate you provide the pytorch code of tree-lstm. Thanks a lot.

**Comment by ghost** [comment]
When I run "fetch_and_preprocess.sh"
there is an error:
can not find file: /lib*.java.
What can I do to fix this problem?
